testing/matrix-sdk-integration-testing/src/tests/sliding_sync/room.rs

```diff
@@ -538,3 +551,354 @@ async fn test_room_notification_count() -> Result<()> {
 
     Ok(())
 }
+
+/// Boolean that decides if to_device messages should be dropped.
+static DROP_TODEVICE: StdMutex<bool> = StdMutex::new(true);
+
+/// Response preprocessor that drops to_device events if DROP_TODEVICE is true.
+fn drop_todevice(response: &mut Bytes) {
+    // Looks for a json payload containing "extensions" with a "to_device" part.
+    // This should only match the sliding sync response. In all other cases, it
+    // makes no changes.
+    let Ok(mut json) = serde_json::from_slice::<serde_json::Value>(response) else {
+        return;
+    };
+    let Some(extensions) = json.get_mut("extensions").and_then(|e| e.as_object_mut()) else {
+        return;
+    };
+    // Remove to_device field if it exists
+    let Some(to_device) = extensions.remove("to_device") else {
+        return;
+    };
+    if *DROP_TODEVICE.lock().unwrap() {
+        info!("Dropping to_device: {to_device}");
+        *response = serde_json::to_vec(&json).unwrap().into();
+        return;
+    }
```

Comment:
I think it might be easier to understand by checking the boolean first, and if it's false, return early.